Verse/src/DecoderDescriptors/Tree/TreeDecoderStream.cs

```diff
@@ -28,9 +28,20 @@ public void Dispose()
 
 		public bool TryDecode(out TEntity entity)
 		{
-			entity = this.constructor();
+			var entityValue = this.constructor();
 
-			return this.callback(this.reader, this.state, ref entity);
+			var result = this.callback(this.reader, this.state, ref entityValue, out var isVoid);

```

Comment:
This looks weird: since we're passing an "entityValue" on which caller has full control, why do we need an additional "isVoid" value that we used to override the value set to "entityValue"? Shouldn't this be done internally by the method?

[update] I think I got the logic behind. The comment above from this review is probably more accurate than this one :)